service/frontend/workflowHandler.go

```diff
@@ -1337,8 +1338,9 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		return nil, err
 	}
 
-	if getRequest.GetMaximumPageSize() == 0 {
-		getRequest.MaximumPageSize = common.Int32Ptr(wh.config.DefaultHistoryMaxPageSize)
+	if getRequest.GetMaximumPageSize() <= 0 {
+		domainFilter := dynamicconfig.DomainFilter(getRequest.GetDomain())
```

Comment:
Can we define a strongly typed helper function on config which takes the domain when reading any config which supports filter on domain.  Then we can have the code to create filter in one location and have a straightforward API for rest of the code to load the config.

Reply:
add GetIntPropertyWithDomainFilter